src/constructs/autoscaling/asg.test.ts

```diff
@@ -0,0 +1,135 @@
+import "@aws-cdk/assert/jest";
+import { SynthUtils } from "@aws-cdk/assert/lib/synth-utils";
+import { Vpc } from "@aws-cdk/aws-ec2";
+import { ApplicationProtocol } from "@aws-cdk/aws-elasticloadbalancingv2";
+import { App, Stack } from "@aws-cdk/core";
+import { GuStack } from "../core/stack";
+import { GuSecurityGroup } from "../ec2";
+import { GuApplicationTargetGroup } from "../loadbalancing";
+import type { GuAutoScalingGroupProps } from "./asg";
+import { GuAutoScalingGroup } from "./asg";
+
+interface SynthedStack {
+  Resources: Record<string, { Properties: Record<string, unknown> }>;
+}
+
+describe("The GuAutoScalingGroup", () => {
+  const vpc = Vpc.fromVpcAttributes(new Stack(), "VPC", {
+    vpcId: "test",
+    availabilityZones: [""],
+    publicSubnetIds: [""],
+  });
+  const defaultProps: GuAutoScalingGroupProps = {
+    vpc,
+    imageId: "123",
+    instanceType: "t3.macro",
```

Comment:
`macro`? Is this intentional or a typo of `micro`?

Reply:
It was intentional just to guarantee that it had used the value from props rather than a default. Happy to change as it looks a bit like a typo though!

Reply:
Fair enough!